integration_tests/steps/two_sided_market.py

```diff
@@ -156,26 +156,26 @@ def energy_origin(context, producer, consumer):
 @then('trades are matched only on the Grid market')
 def trades_matched_on_grid(context):
     # Assert that all grid trades contain Offer objects, bid trades are not tracked
-    assert all(isinstance(trade.offer, Offer)
+    assert all(isinstance(trade.offer_bid, Offer)
                for market in context.simulation.area.past_markets
                for trade in market.trades)
 
     # Assert that all House 1 trades contain Bid objects
     house1 = next(c for c in context.simulation.area.children if c.name == "House 1")
     load = next(c for c in house1.children if c.name == "H1 General Load")
     load_energy_per_day = load.strategy.energy_per_slot_Wh / 1000 * 24
-    house1_consumed_energy = sum(trade.offer.energy
+    house1_consumed_energy = sum(trade.offer_bid.energy
                                  for market in house1.past_markets
                                  for trade in market.trades)
     assert isclose(house1_consumed_energy, load_energy_per_day, rel_tol=1e-3)
 
     # Assert that all House 2 trades contain Offer objects
     house2 = next(c for c in context.simulation.area.children if c.name == "House 2")
-    assert all(isinstance(trade.offer, Offer)
+    assert all(isinstance(trade.offer_bid, Offer)
```

Comment:
Now that all these occurrences were revealed, it would be nice if we got rid of control flow via type checking. How about adding two extra methods/properties in both Offer and Bid classes, named `is_offer` and `is_bid` that will return True / False respectively and will be used for these methods ? I think it would help a lot in the readability of the caller code, if you disagree though feel free to skip it because this is purely according to personal preference.

Reply:
What do you think about adding these interfaces to the Trade class ? maybe named is_offer_trade() and is_bid_trade()

Reply:
Even better! As long as the check is self-contained in the Trade class and not done by the caller code it works for me, thanks!